### PR TITLE
Upgrading the dependencies and versions

### LINKED ISSUE #665: Upgrade all pacakges to last version
Please upgrade all packages to the last version and make sure the app runs without errors

### PR DESCRIPTION
**Pull request by andoriyaprashant**
## Description

I've noticed several available updates and proactively upgraded dependencies. During this process, I conducted thorough testing with the updated dependencies and encountered no errors, which is a marked improvement from previous instances where errors were prevalent. By aligning certain dependencies with their compatible versions, I successfully resolved the previously encountered errors.

## Screenshots of the changes - 

image

- I upgrade dependencies with the help of "flutter pub upgrade --major -version" command

image

- Build gradle 

image
image
image

### CODE REVIEW

**Comment by andoriyaprashant** [comment]
Hello @guyluz11 please review my pull request #679 
Thank you

File: android/build.gradle [hunk 1]
```diff
@@ -1,17 +1,17 @@
 buildscript {
-    ext.kotlin_version = '1.7.20'
+    ext.kotlin_version = '1.7.10'
```
Comment (guyluz11) [inline android/build.gradle:2]: Why Kotlin version down?

**Review by guyluz11 — CHANGES_REQUESTED** [review]

**Comment by guyluz11** [comment]
I understand from what you wrote "improvement from previous instances where errors were prevalent" that you got out of the main branch, please fork and pr to dev next time 
As you can see his last build passed
image

File: android/build.gradle (hunk 1 above)
Comment (andoriyaprashant) [reply android/build.gradle:2]: I might have made a typing mistake in this. I will correct it now

**Comment by andoriyaprashant** [comment]
> I understand from what you wrote "improvement from previous instances where errors were prevalent" that you got out of the main branch, please fork and pr to dev next time As you can see his last build passed image
> 
> Also it looks like it does not pass ci, so also needs to be checked

I'll work from the dev branch next time, my apologies

**Review by guyluz11 — APPROVED** [review]

**Comment by andoriyaprashant** [comment]
Sorry for bothering you @guyluz11  , but could you please add the 'hacktobefest-accepted' label to this pull request
My pull request won't be approved without it
Thank you

**Comment by guyluz11** [comment]
My bad, forgot about it

**Pull request merged**